### ISSUE TITLE
New Feature - Was it really a honeypot?

### ISSUE THREAD

**Issue by ghost** [tsarbuig/WalletHunter#18]
It would be really helpful if the bot could keep track of the coins that were initially labeled as honeypots BUT where the **tracked wallet was still able to sell them later on**. This would indicate that the coin wasn't a honeypot after all. 

This feature would be useful for tweaking strategies. For example, if the tracked wallet is profitable at buying and selling coins that were initially detected to be "honeypots," then the user could decide to disable the honeypot protection mechanism. Disabling the honeypot protection in this case would be desirable because they are overall making profits.  

Specifically, I’m thinking WH could display a message indicating that coin originally labeled as a "honeypot" has been successfully sold. 

It would be even nicer if the TX log could display a message that the tracked wallet sold a honeypot coin and for how much profit.
(I know the TX wallet is pretty much only for coins that the user bought, but for analysis purposes, I think this would be a great feature).

**Comment by AdamB2023** [comment]
This sadly isn't entirely true and valid to fully rely on it.
The issue is, that some coins appear to be not a honeypot while they're using whitelists in encrypted source code.
In this scenario, it will look all fine until you will try to sell and won't be able due to not being the whitelist like scamers wallets.
They're intentionally creating a fake-profitable wallets to follow them and pull into rugs.
As Tsarbuig said, "rugging is an art" 😁

**Comment by ghost** [comment]
> This sadly isn't entirely true and valid to fully rely on it. The issue is, that some coins appear to be not a honeypot while they're using whitelists in encrypted source code. In this scenario, it will look all fine until you will try to sell and won't be able due to not being the whitelist like scamers wallets. They're intentionally creating a fake-profitable wallets to follow them and pull into rugs. As Tsarbuig said, "rugging is an art" 😁

WOW, i'm shocked to hear this. This is news to me and I wonder if I have been a victim of this as well. Do you know if EVERY coin of theirs is a "honeypot" or do they like to mix it up to add to the confusion?

**Comment by ghost** [comment]
Also is there a way to "screen" for whitelist functions?

**Comment by AdamB2023** [comment]
They change it during the trade, which makes it difficult to be completely sure.
I've had a case where I took part of the profit, then couldn't sell the rest because I wasn't whitelisted.
Nowadays, I mostly track addresses that don't buy such coins, but if I notice more cases like that, I will do a contract review.
The one I saw had the whitelist execution encrypted, but if I notice it again, I'll keep an eye on it.
Maybe it's possible to scan such action and make a sell attempt with front running.

**Comment by ghost** [comment]
Yes, i think it would be good if WH could somehow screen for whitelist functions. But i don't know how hard that would be to implement.
